modules/api/src/main/java/org/apache/ignite/table/criteria/Expression.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.table.criteria;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * {@code Expression} defines a general typed expression in a Query instance. The generic type parameter
+ * is a reference to the type the expression is bound to.
+ */
+public interface Expression {
+    /**
+     * Accept the visitor with the given context.
+     *
+     * @param <R> return type
+     * @param <C> context type
+     * @param v visitor
+     * @param context context of visit
+     */
+    <R, C> void accept(CriteriaVisitor<C> v, @Nullable C context);
```

Comment:
`R` is unused. There is no return type.
```suggestion
    <C> void accept(CriteriaVisitor<C> v, @Nullable C context);
```